tests/unit/test_charm.py

```diff
@@ -1,7 +1,274 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
+import logging
+import unittest
+from io import StringIO
+from unittest.mock import Mock, PropertyMock, patch
 
-class TestCharm:
-    def test_dummy_pass(self):
-        assert True
+import yaml
+from ops import testing
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+
+from charm import BASE_CONFIG_PATH, CONFIG_FILE_NAME, NRF_RELATION_NAME, UDMOperatorCharm
+
+logger = logging.getLogger(__name__)
+
+VALID_NRF_URL = "https://nrf:443"
+EXPECTED_CONFIG_FILE_PATH = "tests/unit/expected_udmcfg.yaml"
+
+
+class TestCharm(unittest.TestCase):
+    @patch(
+        "charm.KubernetesServicePatch",
+        lambda charm, ports: None,
+    )
+    def setUp(self):
+        self.maxDiff = None
```

Comment:
Guess we don't need tthis.

Reply:
How about keeping it? There are tests comparing config files, it's pretty handy to have it when working with those. Is there any reason for removing it?

Reply:
The only reason is that we don't need it. If you wanna keep it for better output if tests fail, I'm ok with that. It was just a suggestion.